src/com/oltpbenchmark/DBWorkload.java

```diff
@@ -393,472 +279,322 @@ else if (groupingName.equals("all")) {
-                    arrival=Phase.Arrival.POISSON;
-                
-                // If serial is enabled then run all queries exactly once in serial (rather than
-                // random) order
-                String serial_string;
-                serial_string = work.getString("serial[not(@bench)]", "false");
-                serial_string = work.getString("serial" + pluginTest, serial_string);
-                if (serial_string.equals("true")) {
-                    serial = true;
-                }
-                else if (serial_string.equals("false")) {
-                    serial = false;
-                }
-                else {
-                    LOG.fatal(String.format("Invalid string for serial: '%s'. Serial string must be 'true' or 'false'",
-                            serial_string));
-                    System.exit(-1);
+                Phase.Arrival arrival = Phase.Arrival.REGULAR;
+                String arrive = work.getString("@arrival", "regular");
+                if (arrive.toUpperCase().equals("POISSON")) {
+                    arrival = Phase.Arrival.POISSON;
                 }
 
-                // We're not actually serial if we're running a script, so make
-                // sure to suppress the serial flag in this case.
-                serial = serial && (wrkld.getTraceReader() == null);
+                // We now have the option to run all queries exactly once in
+                // a serial (rather than random) order.
+                boolean serial = Boolean.parseBoolean(work.getString("serial", Boolean.FALSE.toString()));
+
 
                 int activeTerminals;
                 activeTerminals = work.getInt("active_terminals[not(@bench)]", terminals);
                 activeTerminals = work.getInt("active_terminals" + pluginTest, activeTerminals);
+                // If using serial, we should have only one terminal
+                if (serial && activeTerminals != 1) {
+                    LOG.warn("Serial ordering is enabled, so # of active terminals is clamped to 1.");
+                    activeTerminals = 1;
+                }
                 if (activeTerminals > terminals) {
-                    LOG.error(String.format("Configuration error in work %d: "
-                            + "Number of active terminals is bigger than the total number of terminals", i));
+                    LOG.error(String.format("Configuration error in work %d: " + "Number of active terminals is bigger than the total number of terminals", i));
                     System.exit(-1);
                 }
 
                 int time = work.getInt("/time", 0);
                 int warmup = work.getInt("/warmup", 0);
                 timed = (time > 0);
-                if (scriptRun) {
-                    LOG.info("Running a script; ignoring timer, serial, and weight settings.");
-                }
-                else if (!timed) {
+                if (!timed) {
```

Comment:
TODO NIT Seeing these strings + to each other on the same line makes me want to squish them together